### DISCUSSION TITLE
Cross-platform api to open links?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#12260] [Q&A] [answered]
Hi, is there an API to open hyperlinks that work for all platform?
Used to call Process, but that do not work on WASM and that probably only work on Windows.
Thank you.

**Comment by wieslawsoltes** [comment, marked as answer]
Check https://github.com/wieslawsoltes/HackerNews I have IBrowserService and implemented for all platforms

**Reply by dbriard** [reply]
Thanks a lot for your sample, I dit not see the WASM implementation but I get the idea for others platforms, and I should have a look at Xamarin Essential.

**Reply by wieslawsoltes** [reply]
You could inject support using JS interop here https://github.com/wieslawsoltes/HackerNews/blob/main/HackerNews.Browser/AppBundle/main.js

**Reply by wieslawsoltes** [reply]
Example interop https://github.com/wieslawsoltes/ChatGPT/blob/f9c7f2811e6e5a0835d1fb0cd79eb7e4aee0b9de/src/ChatGPT.UI.Browser/AppBundle/main.js#L15
https://github.com/wieslawsoltes/ChatGPT/blob/main/src/ChatGPT.UI.Browser/Services/Interop.cs

**Comment by robloo** [comment]
This is definitely needed in Avalonia itself. It was last discussed with thr HyperlinkButton PR before it got closed.

https://github.com/AvaloniaUI/Avalonia/pull/9385#issuecomment-1308175710

For now each App is implementing it.

**Reply by dbriard** [reply]
Sure it would be handy to have it in Avalonia for things like HyperlinkButton or Hyperlink inline.
